pixelgl/canvas.go

```diff
@@ -37,32 +37,32 @@ func NewCanvas(bounds pixel.Rect) *Canvas {
 		col: mgl32.Vec4{1, 1, 1, 1},
 	}
 
+	baseShader(c)
 	c.SetBounds(bounds)
+	c.shader.update()
+	return c
+}
 
-	var shader *glhf.Shader
-	mainthread.Call(func() {
-		var err error
-		shader, err = glhf.NewShader(
-			canvasVertexFormat,
-			canvasUniformFormat,
-			canvasVertexShader,
-			canvasFragmentShader,
-		)
-		if err != nil {
-			panic(errors.Wrap(err, "failed to create Canvas, there's a bug in the shader"))
-		}
-	})
-	c.shader = shader
+// SetUniform will update the named uniform with the value of any supported underlying
+// attribute variable. If the uniform already exists, including defaults, they will be reassigned
+// to the new value. The value can be a pointer.
+func (c *Canvas) SetUniform(Name string, Value interface{}) {
+	c.shader.SetUniform(Name, Value)
+}
 
-	return c
+// SetFragmentShader allows you to set a new fragment shader on the underlying
+// framebuffer. Argument "fs" is the GLSL source, not a filename.
+func (c *Canvas) SetFragmentShader(fs string) {
```

Comment:
change name of the argument to something more meaningful, for example: `fs` -> `sourceCode`

Reply:
I'd expect the variable name to be `s`